pkg/controller/database/dbsubnetgroup/controller.go

```diff
@@ -18,175 +18,206 @@ package dbsubnetgroup
 	}
 
 	// in a successful response, there should be one and only one object
-	if len(response.DBSubnetGroups) != 1 {
-		return managed.ExternalObservation{}, errors.Errorf(errMultipleItems, cr.Spec.DBSubnetGroupName)
+	if len(res.DBSubnetGroups) != 1 {
+		return managed.ExternalObservation{}, errors.Errorf(errMultipleItems, cr.Spec.ForProvider.DBSubnetGroupName)
 	}
 
-	observed := response.DBSubnetGroups[0]
-
-	cr.SetConditions(runtimev1alpha1.Available())
+	observed := res.DBSubnetGroups[0]
+	current := cr.Spec.ForProvider.DeepCopy()
+	dbsg.LateInitialize(&cr.Spec.ForProvider, &observed)
+	if !reflect.DeepEqual(current, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, errKubeUpdateFailed)
+		}
+	}
+	cr.Status.AtProvider = dbsg.GenerateObservation(observed)
+
+	switch cr.Status.AtProvider.SubnetGroupStatus {
+	case v1beta1.DBSubnetGroupStateAvailable:
+		cr.Status.SetConditions(runtimev1alpha1.Available())
+		resource.SetBindable(cr)
+	case v1beta1.DBSubnetGroupStateCreating:
+		cr.Status.SetConditions(runtimev1alpha1.Creating())
+	case v1beta1.DBSubnetGroupStateDeleting:
+		cr.Status.SetConditions(runtimev1alpha1.Deleting())
+	default:
+		cr.Status.SetConditions(runtimev1alpha1.Unavailable())
+	}
 
-	cr.UpdateExternalStatus(observed)
+	upToDate, err := dbsg.IsDBSubnetGroupUpToDate(cr.Spec.ForProvider, observed)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errUpToDateFailed)
+	}
 
 	return managed.ExternalObservation{
-		ResourceExists:    true,
-		ConnectionDetails: managed.ConnectionDetails{},
+		ResourceUpToDate: upToDate,
+		ResourceExists:   true,
 	}, nil
 }
 
 func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
-	cr, ok := mgd.(*v1alpha3.DBSubnetGroup)
+	cr, ok := mgd.(*v1beta1.DBSubnetGroup)
 	if !ok {
 		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
 	}
 
-	cr.Status.SetConditions(runtimev1alpha1.Creating())
+	cr.SetConditions(runtimev1alpha1.Creating())
 
-	input := &awsrds.CreateDBSubnetGroupInput{
-		DBSubnetGroupDescription: aws.String(cr.Spec.DBSubnetGroupDescription),
-		DBSubnetGroupName:        aws.String(cr.Spec.DBSubnetGroupName),
-		SubnetIds:                cr.Spec.SubnetIDs,
-		Tags:                     []awsrds.Tag{},
+	if cr.Status.AtProvider.SubnetGroupStatus == v1beta1.DBSubnetGroupStateCreating {
+		return managed.ExternalCreation{}, nil
 	}
 
-	for _, t := range cr.Spec.Tags {
-		input.Tags = append(input.Tags, awsrds.Tag{
-			Key:   aws.String(t.Key),
-			Value: aws.String(t.Value),
-		})
+	input := &awsrds.CreateDBSubnetGroupInput{
+		DBSubnetGroupDescription: aws.String(cr.Spec.ForProvider.DBSubnetGroupDescription),
+		DBSubnetGroupName:        aws.String(meta.GetExternalName(cr)),
+		SubnetIds:                cr.Spec.ForProvider.SubnetIDs,
+		Tags:                     []awsrds.Tag{},
```

Comment:
An empty array of tags is sent here. A `for` similar to the existing one would do the job.